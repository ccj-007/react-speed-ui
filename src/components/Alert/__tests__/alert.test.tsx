import { render, fireEvent, screen } from '@testing-library/react';
import { Alert, Button } from '../../index';
import "@testing-library/jest-dom/extend-expect";
import React from 'react';

describe('test alert component', () => {
  it('should render', () => {
    const { asFragment } = render(
      <>
        <Alert
          title='title content'
          alertType='success'
          content='test content'
        ></Alert>
      </>
    );
    expect(asFragment()).toMatchSnapshot();
  })

  it('click it show alert', () => {
    const Demo = () => {
      let [isOpen, setIsOpen] = React.useState(false);
      render(
        <>
          <Button onClick={() => setIsOpen(true)}>
            alert
          </Button>
          <Alert
            isOpen={isOpen}
            title='title content'
            alertType='success'
            content='test content'
            closeAlert={() => setIsOpen(false)}
          ></Alert>
        </>
      )
    }
    const { getByTestId } = render(
      <Demo />
    );
    fireEvent.click(screen.getByText(/alert/i))
    expect(getByTestId('alert')).toBeInTheDocument()
    expect(getByTestId('alert')).toHaveClass('speed-alert speed-alert-success')
  });
})
