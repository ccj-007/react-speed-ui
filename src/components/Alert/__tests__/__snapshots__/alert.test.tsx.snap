// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test alert component should render 1`] = `<DocumentFragment />`;
